dev-docs/source/VSCode.rst

```diff
@@ -0,0 +1,325 @@
+
+Required
+--------
+- `C/C++ <https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools>`_
+    - Required for C++ linting, debugging etc
+- `Python <https://marketplace.visualstudio.com/items?itemName=ms-python.python>`_
+    - Required to run any python code, debug etc
+
+Recommended
+-----------
+- `Clang-Format <https://marketplace.visualstudio.com/items?itemName=xaver.clang-format>`_
+    - Allows auto formatting using the clang format version installed on your machine
+- `CMake <https://marketplace.visualstudio.com/items?itemName=twxs.cmake>`_
+    - Allows language highlighting and auto fill for CMake
+- `Doxygen Document Generator <https://marketplace.visualstudio.com/items?itemName=cschlosser.doxdocgen>`_
+    - Once a function is written with this extension it can generate parts of the doxygen comments itself
+- `GitLens <https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens>`_
+    - Very powerful source control assistant
+- `Visual Studio Intellicode <https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.vscodeintellicode>`_
+    - This improves your code suggestions in certain languages
+
+Features
+========
+
+Auto Formatting
+---------------
+With a formatter installed as an extension it is recommended to use Clang-Format. It
+is possible to format your code using on save, type, paste and save timeout. To set
+when you want to format:
+- Open the File->Preferences->Settings menu
+- Search for "Format On"
+- Then tick the times at which formatting should occur. (recommend save only)
+
+Auto Saving
+-----------
+This is a built in feature for VSCode much like with formatting however no extension
+is required. You can save on window change, on focus change, and on delay.
+- Open the File->Preferences->Settings menu
+- Search for "Auto Save"
+- Select the auto save type that is wanted (recommend `onFocusChange`)
+
+Building
+--------
+This is where VSCode comes into its own as a versatile environment for programming
+in any language, because of this it has a Tasks list. These tasks are saved in
+tasks.json inside the .vscode folder now present in any opened folder/workspace.
+
+- To start creating a task open the command line (Ctrl+Shift+P or ⌘+Shift+P)
+- Type "Task"
+- Select "Task: Configure Task" by navigating using the arrow keys
+- Select the first option that is in the menu and it should open tasks.json
+
+The tasks.json allows the creation of Build Tasks default or otherwise. VSCode will
+attempt to make a useful task for you. However it may be better to use this example:
+
+**Linux/OSX:**
+
+This assumes that you have generated in CMake using ninja. The commands can be
+switched out with the command and various args for whatever you have generated with.
+
+.. code-block:: javascript
+
+    {
+        "version": "2.0.0",
+        "tasks": [
+            {
+                "label": "Build Mantid",
+                "type": "shell",
+                "command": "ninja",
```

Comment:
```suggestion
                "command": "cmake --build .",
```

This makes the Ninja assumption above irrelevant.

Reply:
I don't agree with this change, as we should be using ninja as our build system on MacOS or Linux.

Reply:
The above call will use Ninja if that's the generator you specified. `cmake --build .` will always work so should be preferred here.